app/service/atm_service.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+require 'singleton'
+require_relative '../repo/atm_repository'
+require_relative '../repo/bank_repository'
+require_relative '../models/response'
+require_relative '../models/transaction'
+require_relative '../models/user'
+require_relative '../models/atm'
+
+class ATMService
+  include Singleton
+
+  MAX_WITHDRAW_PER_DAY = 5000
+
+  def initialize
+    @@bank_repository = BankRepository.instance
+    @@atm_repository = ATMRepository.instance
+  end
+
+  def deposit(atm_id, amount, email)
+    if amount <= 0
+      Error.new("Amount must be greater than 0")
+    else
+      @@bank_repository.change_balance(email, amount)
+      @@bank_repository.register_transaction(email, Transaction.new(amount, TransactionTypes::DEPOSIT))
+      @@atm_repository.change_total_cash(atm_id, amount)
+      Success.new("Amount deposited successfully")
+    end
+  end
+
+  def withdraw(atm_id, amount, email)
+    atm = @@atm_repository.get_atm(atm_id)
+    user_founds = @@bank_repository.get_balance(email)
+    total_withdrawn = @@bank_repository.get_total_withdrawn_today(email)
+    return Error.new("Amount must be greater than 0") if amount <= 0
+    return Error.new("This ATM has insufficient funds") if atm.total_cash < amount
+    return Error.new("Insufficient funds in your account") if user_founds < amount
+    return Error.new("You have exceeded the maximum withdrawal amount per day") if total_withdrawn + amount > MAX_WITHDRAW_PER_DAY
+
```

Comment:
I would extract this in a separate method or even a class that's performing these validations